src/backend/kvm/thread.rs

```diff
@@ -117,37 +101,67 @@ impl<P: KeepPersonality> super::super::Thread for Thread<P> {
                     .unwrap();
 
                 // If some other thread tried to use the same block, the above unwrap would have panicked.
-                let block = unsafe { &mut *block_virt.as_mut_ptr::<Block>() };
-
-                // To avoid clashing of rep and req in the union, clone the request
-                let req = unsafe { block.msg.req };
-
-                let ret = match i64::from(req.num) {
-                    SYS_ENARX_BALLOON_MEMORY => {
-                        let rep = self.balloon(&req).into();
-                        block.msg.rep = rep;
-                        Ok(Command::Continue)
+                let block: Block = unsafe {
+                    &mut *block_virt.as_mut_ptr::<[usize; BLOCK_SIZE / size_of::<usize>()]>()
+                }
+                .as_mut_slice()
+                .into();
+
+                sallyport::host::execute(block.into_iter().filter_map(|item| {
+                    if cfg!(feature = "dbg")
+                        && !matches!(&item, Item::Syscall(syscall, ..) if syscall.num == 1)
```

Comment:
What exactly does this do? Why skip `write`? Also why not use `libc::SYS_*` constant?

Reply:
That would then debug the debug messages also :)
yes to `libc::SYS_*` .. it was a quick solution, not cleaned up yet.
Pointing out the dirty areas, which need work is perfect. Thank you!

Reply:
fixed and completely rewritten